### DISCUSSION TITLE
Dynamic mask based on another field value?

### DISCUSSION THREAD

**Discussion by maxacarvalho** [filamentphp/filament#10395] [General]
Hi 👋🏽 

I'm trying to apply a mask based on another field. It's a monetary value, so depending on the currency the mask should change. For example, in Brazil the mask for one thousand and fifty cents is `1.000,50`, while in the US is `1,000.50`.

I have a **Select** field to choose the currency. Whenever I change the currency the amount field updates itself but stops behaving as expected.

### Without selecting the currency
<img width="471" alt="image" src="https://github.com/filamentphp/filament/assets/45166618/a96f1dd9-4680-4b64-9056-e05f4ddec676">

### After selecting the currency
<img width="503" alt="image" src="https://github.com/filamentphp/filament/assets/45166618/c3b17fa2-cffc-4190-84a2-f8f54a9e4542">

### And on top of that, it won't allow me to enter more than three digits
<img width="475" alt="image" src="https://github.com/filamentphp/filament/assets/45166618/be4a57ef-2af0-4d08-be96-e2aad7288ca3">

Here's the code (snippet)
```php
Select::make('last_price_currency')
    ->label(Str::title(__('product.last_price_currency')))
    ->options(fn () => Currency::query()->distinct()->pluck(Currency::ISO_CODE, Currency::ISO_CODE)->toArray())
    ->required()
    ->live(),

TextInput::make(Product::LAST_PRICE)
    ->label(Str::title(__('product.last_price')))
    ->required()
    ->formatStateUsing(function ($state, Model|Product|null $record, Get $get) {
        $currency = null === $record ? $get('last_price_currency') : $record->last_price->currency;

        if (null === $record) {
            return 'BRL' === $currency ? '0,00' : '0.00';
        }

        return $record->last_price->getFormattedAmount();

    })
    ->mask(function (TextInput $component, Model|Product|null $record, Get $get) {
        $currency = null === $record ? $get('last_price_currency') : $record->last_price->currency;

        if ('BRL' === $currency) {
            return RawJs::make('$money($input, \',\', \'.\')');
        }

        return RawJs::make('$money($input)');
    }),
```

**Comment by rtrudel** [comment]
I have the same problem, I would like to change the postal code mask to `a9a 9a9` when my country is `CA` and let it unmasked if not.
